course_access_groups/models.py

```diff
@@ -52,6 +56,36 @@ class MembershipRule(utils_models.TimeStampedModel):
     domain = models.CharField(max_length=255, db_index=True, help_text='The learner email domain e.g. "example.com".')
     group = models.ForeignKey(CourseAccessGroup, on_delete=models.CASCADE)
 
+    @classmethod
+    def apply_for_user(cls, user):
```

Comment:
It looks like this method could be a model manager method instead of a model class method, especially since it it interacting with multiple models

Reply:
It was 50/50 for me, and I chose `MembershipRule`. Now moved to `Membership`. Thanks John for breaking the tie.

Reply:
Thanks Omar!